File: API/OnlyFive.Types/Models/Game.cs
using OnlyFive.Types.Core.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OnlyFive.Types.Models
{
    public class Game
    {
        public int Id { get; set; }
        public string UrlId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string HostDevice { get; set; }
        public string GuestDevice { get; set; }
        public string HostId { get; set; }
        public string GuestId { get; set; }
        public int HostScore { get; set; }
        public int GuestScore { get; set; }
        public GameRoundsEnum GameRound   { get; set; }

        public virtual ApplicationUser Host { get; set; }
        public virtual ApplicationUser Guest { get; set; }
        public virtual ICollection<Round> Rounds { get; set; }
    }
}

File: SPA/src/Types/Game.ts
import { Pin } from "src/tools/pin";
import { Player } from "src/tools/player";

export class Game {
    id: number;
    urlId: string;
    startDate: Date;
    endDate?: Date;
    hostDevice: string;
    guestDevice: string;
    hostId: string;
    guestId: string;
    gameRound: GameRoundsEnum;
    host: Player;
    guest: Player;
    hostName: string;
    guestName: string;
    hostScore: number;
    guestScore: number;
    lastRoundOffset: number;
}
export class Round {
    offset: number;
    startDate: Date;
    endDate: Date;
    pawnMap: string;
    gameId: number;
}
export class LastRound {
    round: Round;
    game: Game;
}

export class NewPin {
    offset: number;
    pin: Pin;
    gameUrlId: string;
}

export class CompleteRound {
    offset: number;
    playerId: string;
    gameUrlId: string;
}

export enum GameRoundsEnum{
    One = 1,
    Three = 3,
    Five = 5,
    Nine = 9
}
